doc/man/tracef-tracelog-limitations.txt

```diff
@@ -0,0 +1,21 @@
+The +{macro-name}()+ utility macro was developed to make user space
+tracing super simple, albeit with notable disadvantages compared to
+custom, full-fledged tracepoint providers:
+
+  * All generated events have the same provider/event names.
+  * There's no static type checking.
+  * The only event field with user data you actually get, named `msg`,
+    is a string potentially containing the values you passed to the
+    macro using your own format. This also means that you cannot use
+    filtering using a custom expression at run time because there are no
+    isolated fields.
+  * Since +{macro-name}()+ uses C standard library's man:vasprintf(3)
+    function in the background to format the strings at run time, its
+    expected performance is lower than using custom tracepoint providers
+    with typed fields, which do not require a conversion to a string.
```

Comment:
It is also not as compact in the traces to save a string rather than binary (generally).